openmc/material.py

```diff
@@ -1210,6 +1230,59 @@ def mix_materials(cls, materials, fracs: typing.Iterable[float],
 
         return new_mat
 
+    @classmethod
+    def from_library(cls, name, library='pnnl_v2'):
+        """Adds a material composition from a predefined library
+
+        Parameters
+        ----------
+        name : str
+            Reference name of the material (case sensitive)
+        library : str
+            The material library to use. Options include: 'pnnl_v2' The Pacific
+            Northwest National Laboratory Compendium of Material Composition
+            Data for Radiation Transport Modeling, available from
+            https://www.pnnl.gov/main/publications/external/technical_reports/PNNL-15870Rev2.pdf
+
+        """
+
+        if library not in MATERIAL_LIBRARIES.keys():
+            msg = (
+                    f'library {library} not found in available libraries. '
+                    f'Supported libraries are {list(MATERIAL_LIBRARIES.keys())}'
+            )
+            raise ValueError(msg)
+            
+        global _MATERIAL_LIBRARY_CACHE
+
+        # loads in the library into the MATERIAL_LIBRARIES if not already loaded
+        try:
+            library_data = _MATERIAL_LIBRARY_CACHE[library]
+        except KeyError:
+            library_data = json.loads(MATERIAL_LIBRARIES[library].read_text())
```

Comment:
It looks like we are loading in the material library from the JSON file "as-is". This is fine w/ me but it means we expect those JSON files to have a particular structure -- did we document this somewhere for users who want to create a their own, custom library?

Reply:
I would probably personally prefer to have a well defined material library format in XML for consistency with other data (especially materials). We could almost just use the material XML format but ignore ID numbers. Although there is the complication of not expanding elements so it is a bit different, but for validating a spec and roundtrip consistency I might go with XML - thoughts?

Reply:
XML files would mean we can use the ```Material.from_xml_element``` method if the materials are defined entirely as nuclides (no elements)

Reply:
PR is now using xml files as suggested